### PR TITLE
Do not allow VaultAdmin implementation to be set to a non-contract

### PR DESCRIPTION
**Pull request by DanielVF**
If you made a contract change, make sure to complete the checklist below before merging it in master.

Refer to our [documentation](https://github.com/OriginProtocol/security) for more details about contract security best practices.

Contract change checklist:
  - [x] Code reviewed by 2 reviewers
  - [x] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [x] Echidna tests pass (not automated, run manually on local)

### CODE REVIEW

**Comment by DanielVF** [comment]
# Contract review checklist

Reviewer: Daniel Von Fange

#### Tests

[x] Each logical branch has a test

#### Overflow

- [x] Never use "+" or "-", always use safe math

#### Black magic

- [x] Does not contain `selfdestruct`
- [x] Does not use `delegatecall` outside of proxying

#### Authentication

_No changes to authentication_

#### Cryptographic code

_No cryptographic code_

#### Gas problems

_No loops_

#### External calls

_No external calls_

#### Ethereum

[x] Contract does not send or receive Ethereum.

File: contracts/test/vault/upgrade.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    vault = fixture.vault;
+    ousd = fixture.ousd;
+    governor = fixture.governor;
+    vaultStorage = await hre.ethers.getContractAt(
+      "VaultStorage",
+      vault.address
+    );
+  });
+
+  it("should upgrade to a new admin implimentation", async function () {
```
Comment (tomlinton) [inline contracts/test/vault/upgrade.js:19]: Typo here and in the below test

File: contracts/contracts/vault/VaultStorage.sol [hunk 2]
```diff
@@ -98,6 +99,10 @@ contract VaultStorage is Initializable, Governable {
      * @param newImpl address pf the implementation
```
Comment (tomlinton) [inline contracts/contracts/vault/VaultStorage.sol:99]: Typo pf

**Review by tomlinton — APPROVED** [review]

File: contracts/test/vault/upgrade.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    ousd = fixture.ousd;
+    governor = fixture.governor;
+    vaultStorage = await hre.ethers.getContractAt(
+      "VaultStorage",
+      vault.address
+    );
+  });
+
+  it("should upgrade to a new admin implementation", async function () {
+    const newVaultImpl = ousd.address; // ;)
```
Comment (franckc) [inline contracts/test/vault/upgrade.js:20]: nit: clean up end empty comment

**Review by franckc — APPROVED** [review]
Thanks for adding this defensive code.

Do you mind adding a deployment script for it? Feel free to ping me and I can help...

File: contracts/test/vault/upgrade.js (hunk 3 above)
Comment (DanielVF) [reply contracts/test/vault/upgrade.js:20]: Lol, I failed at humor.

The comment is supposed to represent  a winking face at the absurdity of setting the vault implementation address to the OUSD proxy address. 😄

Comment (DanielVF) [reply contracts/test/vault/upgrade.js:20]: I'm fine waiting for this change to be rolled into future upgrades.

Comment (franckc) [reply contracts/test/vault/upgrade.js:20]: I'm a code review robot. I do not understand humor from you, human // :)

**Pull request merged**